src/todo/todo.lua

```diff
@@ -107,7 +111,7 @@ function todo.get_task_from_add_frame(frame)
 
     local should_add_to_top = false
     -- 'Add to Top' control won't exist in an edit dialog
-    local add_top_control = frame.todo_add_task_table.children[5]
+    local add_top_control = frame.todo_add_task_table["todo_add_top"]
```

Comment:
This was needed to fixe a crash when updating a task, the "todo_add_top" component did not exists when updating a task.

Reply:
This is interesting. This never happened in my tests.
Might this be a slightly different behaviour in different versions of Factorio? I was on 16.0 I think...

Reply:
I'm on 0.16.51 here. This crashed everytime. Seems like the children[5] was not always the checkbox, as there is no checkbox in edit mode, only in add mode.